### PR TITLE
Add the Debugging packages section

### PR DESCRIPTION
**Pull request by kwitaszczyk**

### CODE REVIEW

File: src/packages/debugging.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+Alternatively, you can selectively reinstall a package:
+
+```
+pkg64c upgrade -r CheriBSD-debug <pkg-name>
+```
+
+However, keep in mind that in such case only the package you list is
+reinstalled, leaving its dependencies intact.
```
Comment (brooksdavis) [inline src/packages/debugging.md:41]:
```suggestion
reinstalled, leaving its dependencies including libraries intact and without debug symbols.
```

File: src/packages/debugging.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Once you enable the `CheriBSD-debug` repository, remember to always specify
+which package repository you want to use with the `-r` command flag when
+executing a `pkg64c` command, e.g.
+`pkg64c install -r CheriBSD <pkg-name>` or
+`pkg64c install -r CheriBSD-debug <pkg-name>`.
+Otherwise, it can happen that `CheriBSD-debug` is used whilst you want to use
+the default package repository.
```
Comment (brooksdavis) [inline src/packages/debugging.md:49]: Is this true? I though `CheriBSD` has precedence?

Comment (brooksdavis) [reply src/packages/debugging.md:49]: Also, do we want to give an example of disabling `CheriBSD` entirely?

File: src/packages/debugging.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Debugging packages
+
+Default CheriBSD package repositories are built with optimisation levels defined
+in CheriBSD ports or upstream third-party software projects.
+Most ports use the `-O2` optimisation level but some switch to `-O3` for better
+performance or `-O1` due to compiler issues preventing them from using
+a higher optimisation level.
+Unfortunately, the FreeBSD ports build system does not allow to build ports with
+optimisations and debug symbols at the same time.
```
Comment (rwatson) [inline src/packages/debugging.md:9]: I wonder if, in this paragraph, we should reduce a little bit the detail on optimisation and increase the level of detail regarding needing debug symbols to use GDB to debug package crashes?

I wonder if here we should more clearly draw the link between ports and packages, as non-FreeBSD folk may not know that ports build packages. E.g., “.. to build packages with ...”?

File: src/packages/debugging.md (hunk 2 above)
Comment (kwitaszczyk) [reply src/packages/debugging.md]: > Is this true? I though `CheriBSD` has precedence?

I remember I was getting unexpected results when working with two package repositories enabled at the same time, even if the CheriBSD-debug repository has the priority 1, which makes it less important than the CheriBSD repository with the priority 0.

However, I wasn't able to reproduce this issue now. I've rephrased this paragraph to assume that the CheriBSD repository has precedence here, and I can get back to documenting a potential issue once I can reproduce it.

Comment (kwitaszczyk) [reply src/packages/debugging.md]: > Also, do we want to give an example of disabling `CheriBSD` entirely?

That's a good idea. I've added a paragraph on that.

File: src/packages/debugging.md (hunk 3 above)
Comment (kwitaszczyk) [reply src/packages/debugging.md:8]: > I wonder if, in this paragraph, we should reduce a little bit the detail on optimisation and increase the level of detail regarding needing debug symbols to use GDB to debug package crashes?

Yes, I think it makes sense to focus on the debugging story here.

> I wonder if here we should more clearly draw the link between ports and packages, as non-FreeBSD folk may not know that ports build packages. E.g., “.. to build packages with ...”?

I agree.

I've updated the paragraph to fix both issues. Does it look better now?

**Comment by kwitaszczyk** [comment]
Let's merge it for now and we can improve the text if needed.

**Pull request merged**